vespalib/src/vespa/vespalib/btree/btreeiterator.hpp

```diff
@@ -774,28 +768,28 @@ linearSeek(const KeyType & key, CompareT comp)
 
 template <typename KeyT, typename DataT, typename AggrT, typename CompareT,
           typename TraitsT>
+template <bool STRICT_SEEK>
 void
 BTreeConstIterator<KeyT, DataT, AggrT, CompareT, TraitsT>::
 seekPast(const KeyType & key, CompareT comp)
 {
     if (TraitsT::BINARY_SEEK) {
-        binarySeekPast(key, comp);
+        binarySeekPast<STRICT_SEEK>(key, comp);
     } else {
-        linearSeekPast(key, comp);
+        linearSeekPast<STRICT_SEEK>(key, comp);
     }
 }
 
 template <typename KeyT, typename DataT, typename AggrT, typename CompareT,
           typename TraitsT>
+template <bool STRICT_SEEK>
 void
 BTreeConstIterator<KeyT, DataT, AggrT, CompareT, TraitsT>::
 binarySeekPast(const KeyType & key, CompareT comp)
 {
     const LeafNodeType *lnode = _leaf.getNode();
     uint32_t lidx = _leaf.getIdx();
-#ifdef STRICT_BTREE_ITERATOR_SEEK
-    assert(_leaf.valid() && !comp(key, lnode->getKey(lidx)));
-#endif
+    assert( ! STRICT_SEEK || (_leaf.valid() && !comp(key, lnode->getKey(lidx))));
```

Comment:
Might as well remove the assert, instead of adding a template argument to the method.
Or add "assert_before" and "assert_not_after" methods on the iterator that a conservative caller can call.